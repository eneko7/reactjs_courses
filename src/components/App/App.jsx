import React from 'react';
import { Switch, Route } from 'react-router-dom';
import MoviesPage from '../../pages/MoviesPage';
import MovieDetailsPage from '../../pages/MovieDetailsPage';
import PageNotFound from '../../pages/PageNotFound';
import styles from './App.scss';

const App = () => (
  <main className={styles.main}>
    <Switch>
      <Route exact path="/" component={MoviesPage} />
      <Route exact path="/search" component={MoviesPage} />
      <Route exact path="/movies" component={MoviesPage} />
      <Route exact path="/movie/:id" component={MovieDetailsPage} />
      <Route path="*" component={PageNotFound} />
    </Switch>
  </main>

);

export default App;
