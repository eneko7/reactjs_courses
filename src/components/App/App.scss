@font-face {
  font-family: "ProximaNovaBold";
  src: url('../../assets/fonts/ProximaNova-Bold.woff');
}

@font-face {
  font-family: "ProximaNovaRegular";
  src: url('../../assets/fonts/ProximaNova-Regular.woff');
}

body {
  margin: 0;

  .main {
    width: 100%;
    height: 100%;
  }
}
