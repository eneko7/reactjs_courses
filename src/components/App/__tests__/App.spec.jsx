import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';

const renderer = new ShallowRenderer();
renderer.render(<App />);
describe('App', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
