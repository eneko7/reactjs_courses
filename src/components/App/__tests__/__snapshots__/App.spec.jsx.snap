// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<main
  className="main"
>
  <ErrorBoundary>
    <Header />
    <ResultLine />
    <MoviesGrid />
    <Footer />
  </ErrorBoundary>
</main>
`;
